### PR TITLE
Eobs 326 & 327

### PR DESCRIPTION
**Pull request by moderatemisbehaviour**
Contains stories for both [EOBS-326](https://jupiter.bjss.com/browse/EOBS-326) and [EOBS-327](https://jupiter.bjss.com/browse/EOBS-327).

### 326
* Can toggle observations on and off for a patient.
* Warning given when there are open escalation tasks for the patient.

### 327
* Can select a reason for the monitoring exception from a drop down menu when toggling observations.

### CODE REVIEW

File: nh_eobs/models/nh_clinical_patient_monitoring_exception.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    A 'patient monitoring exception reason' is any reason why a patient is
+    currently not being monitored (i.e. regular observations are not being
+    performed).
+    """
+    _name = 'nh.clinical.patient_monitoring_exception.reason'
+
+    display_text = fields.Char(string='Reason')
+    display_model = fields.Many2one(
+        comodel_name='nh.clinical.patient_monitoring_exception.select_reason'
+    )
```
Comment (moderatemisbehaviour) [inline nh_eobs/models/nh_clinical_patient_monitoring_exception.py]: Delete this.

File: nh_eobs_mental_health/models/nh_clinical_wardboard.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        patient_monitoring_exception_activity.__dict__.pop('_ids')
+
+        patient_monitoring_exception_activity.complete(
+            self.env.cr, self.env.uid,
+            patient_monitoring_exception_activity_id
+        )
+
+        self.toggle_obs_stop_flag(spell_id)
+
+    def toggle_obs_stop_flag(self, cr, uid, spell_id, context=None):
```
Comment (moderatemisbehaviour) [inline nh_eobs_mental_health/models/nh_clinical_wardboard.py]: Change to `set_obs_stop_flag`

File: nh_eobs_mental_health/models/nh_clinical_wardboard.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        activity_model = self.pool['nh.activity']
+        escalation_task_domain = [
+            ['data_model', 'like', 'nh.clinical.notification.%'],
+            ['state', 'not in', ['completed', 'cancelled']],
+            ['spell_activity_id', '=', spell_activity_id]
+        ]
+        return any(activity_model.search(
+            cr, uid, escalation_task_domain, context=context))
+
+# class PatientMonitoringException(orm.TransientModel):
```
Comment (moderatemisbehaviour) [inline nh_eobs_mental_health/models/nh_clinical_wardboard.py]: Delete this.

File: nh_eobs_mental_health/models/nh_eobs_api.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            ['patient_id', 'in', patient_id],
+            ['state', 'not in', ['completed', 'cancelled']]
+        ], context=context)
+        if spell_id:
+            spell_id = spell_id[0]
+            obs_stopped = spell_model.read(
+                cr, uid, spell_id, ['obs_stop'], context=context) \
+                .get('obs_stop')
+            if obs_stopped:
+                spell_model.write(
```
Comment (moderatemisbehaviour) [inline nh_eobs_mental_health/models/nh_eobs_api.py]: Should use new `end_patient_monitoring_exception` method.

**Comment by Gimpneek** [comment]
:shipit: away :shipit: away :shipit: away now, :shipit: away :shipit: away :shipit: now

**Comment by moderatemisbehaviour** [comment]
Is that this? https://www.youtube.com/watch?v=Mr_uHJPUlO8

File: nh_eobs_mental_health/views/wardboard_view.xml [hunk 5]
```diff
@@ -29,7 +29,7 @@
                         <img src="/web/static/src/img/warning.png"/>One or more escalation tasks for <field name="patient_name" class="oe_inline"/> are not completed.
                     </span>
                     <h4>Select a reason for observation status change</h4>
-                    <field name="reasons" widget="selection"/>
+                    <field name="reasons" widget="selection" required="1"/>
```
Comment (moderatemisbehaviour) [inline nh_eobs_mental_health/views/wardboard_view.xml]: Add some server-side validation too.

**Comment by Gimpneek** [comment]
:shipit: post 423 amendments

**Comment by Nayira** [comment]
Tests for 326, 327, 410, 423 and 444 have passed. :shipit:

**Pull request merged**